feat(navigation): add pinning for sidebar menu items

ku5ic opened the pull request:
# Feature: Pinned Navigation Items

**Related JIRA Story:** [WEB-5364](https://webiny.atlassian.net/browse/WEB-5364)

## Summary

This PR introduces the ability for users to pin and unpin navigation menu items to a personalized "Pinned" section at the top of the sidebar.  
The feature improves navigation efficiency, supports persistent user preferences, and provides clear visual feedback for pin and unpin actions.

<details>
<summary>SCREENSHOTS</summary>

<img width="262" height="1353" alt="Screenshot 2025-10-14 at 12 34 33" src="https://github.com/user-attachments/assets/dea798b6-256a-421d-9475-f139c9a28ce6" />
<img width="49" height="1348" alt="Screenshot 2025-10-14 at 12 34 52" src="https://github.com/user-attachments/assets/8e09a035-8598-453d-93b2-b0ea9818a087" />


</details>

---

## Changes

- **Pin / Unpin functionality**
  - Added a pin icon to all main and submenu items (visible on hover).
  - Clicking the icon pins the item to a new "Pinned Section" at the top of the navigation.
  - Items can be unpinned either from their source or directly from the pinned section.
  - Original items remain in place with a pinned indicator.

- **Pinned Section**
  - Always visible at the top of the navigation.
  - Displays items in the order they were pinned (FIFO).
  - Shows an unpin icon on hover for quick removal.

- **Persistence**
  - User’s pinned items are saved in their Local Storage.

- **UI & Feedback**
  - Dynamic pin icon states (outlined vs filled) for clarity.
  - Immediate visual updates on pin and unpin.
  - Smooth user experience without reloads.

---

## How to Test

1. Hover over any main or submenu item and click the pin icon.  
   The item should appear in the pinned section at the top.
2. Refresh or log out/in.  
   Pinned items should remain visible.
3. Hover over a pinned item and click the unpin icon.  
   The item should be removed from the pinned section but remain in its original location.
4. Verify that pinned items appear in the order they were pinned.

---

## Out of Scope

- Drag-and-drop reordering of pinned items (to be considered in a future iteration).

---

## Checklist

- [x] Pin and unpin functionality implemented  
- [x] Pinned section added to navigation  
- [x] Persistence integrated via  Local storage  
- [x] UI feedback for pin and unpin actions  
- [x] Tested across sessions  
- [x] Code linted, formatted, and reviewed

ku5ic committed:
feat(navigation): add pinning for sidebar menu items

Introduced ability to pin/unpin sidebar menu items in admin UI navigation.
Added PinnableMenuItem and PinnedMenuItems components. Updated MenuConfig
and related components to support the `pinnable` property. Enabled pinning
for CMS content models and groups. Updated Properties to handle `pinnable`
attribute.

Refs: WEB-5364

ku5ic committed:
refactor(nav): rename pinnableItems to pinnableStates

Renamed the variable pinnableItems to pinnableStates for clarity and
consistency in the PinnedMenuItems component. This improves code
readability by better reflecting the data structure's purpose.

ku5ic committed:
Merge branch 'next' into ku5ic/feature/WEB-5364/menu_items_pinning_functionality

ku5ic committed:
feat(ui): show pin icon on hover and when pinned

Pin icon in menu items is now visible on hover or when the item is pinned.
Pinned menu items are rendered with PinnableMenuItem for consistent UX.

Refs: WEB-5364

ku5ic committed:
feat(ui): simplify pinnable menu item icon logic

Remove unused PinOffIcon and always display PinIcon for pinnable menu items.
Add detailed JSDoc to PinnableMenuItem for improved developer clarity. The
pin/unpin action now uses a single icon, streamlining UI consistency.

Refs: WEB-5364

ku5ic committed:
feat(admin-ui): improve pinned menu items rendering logic

Refactored PinnedMenuItems to optimize filtering and rendering of pinned
menu items. Now uses useMemo for pinnable menus and keys, and only
renders the "Pinned" group if there are pinned items. Enhanced
documentation and clarified prop usage. This improves performance and
code readability.

Refs: WEB-5364

ku5ic committed:
feat(menu): add pinnable option to settings menus

Added the `pinnable={true}` prop to various settings-related `Menu` components
across multiple modules. This enables users to pin these menu items for easier
access and improved navigation within the admin interface.

Refs: WEB-5364

SvenAlHamad commented:
On comment...when pinned there should be a "Pinned" section and a "Webiny" section. See screenshot. 
<img width="2546" height="1572" alt="CleanShot 2025-10-07 at 10 12 39@2x" src="https://github.com/user-attachments/assets/b5b3774e-c18a-4401-a86f-4b6a203e3b45" />

ku5ic committed:
feat(admin-ui): persist pinned menu order in localStorage

Refactored pinnable menu item logic to store both pinned state and order in
localStorage. Added hooks and utility functions for managing pinned menu items
and their order. Updated PinnedMenuItems to display pinned items in user-defined
order. Improved documentation and code clarity.

Refs: WEB-5364

ku5ic committed:
feat(admin-ui): add "Webiny" group to pinned menu items

Added a new Menu.Group labeled "Webiny" to the PinnedMenuItems component.
This provides a dedicated section for Webiny-related menu items in the admin
UI navigation, improving organization and clarity for users.

Refs: WEB-5364

ku5ic commented:
> On comment...when pinned there should be a "Pinned" section and a "Webiny" section. See screenshot. <img alt="CleanShot 2025-10-07 at 10 12 39@2x" width="2000" height="1572" src="https://private-user-images.githubusercontent.com/3808420/498182607-b5b3774e-c18a-4401-a86f-4b6a203e3b45.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk4MzIyNDMsIm5iZiI6MTc1OTgzMTk0MywicGF0aCI6Ii8zODA4NDIwLzQ5ODE4MjYwNy1iNWIzNzc0ZS1jMThhLTQ0MDEtYTg2Zi00YjZhMjAzZTNiNDUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMDdUMTAxMjIzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzE3ODUwZGY4NWU3N2FiODRlM2FkZjJiZTJmYTVkNGNkN2M5ZTg0MDQ1OWYzNjMyNTNhNjdhYmNlZDI3MGY3MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.7wyY6gKXZaugfD4CaO67VIjvpfjAunWnjHcnWFnrfUk">

Added with fca3d7b584c5dd3830a40365710d963ac9af4ee1

ku5ic committed:
Merge branch 'next' into ku5ic/feature/WEB-5364/menu_items_pinning_functionality

SvenAlHamad commented:
Based on the last video, you're missing the icon on the pinned entries. Every entry that's pinned should have an icon. The icon you need to use it the icon from the parent group where that entry is placed.

Pavel910 commented on packages/app-admin-ui/src/Navigation/PinnableMenuItem.tsx:
```diff
@@ -0,0 +1,120 @@
+ * @param name - The unique name of the menu item.
+ * @returns The localStorage key string for the pinned state.
+ */
+export const PINNED_KEY = (name: string) => `navigation/${name}/pinned`;
```
This type of case is reserved for constants, like you have `PINNED_ORDER_KEY` later in the code. If it's a function, export it as `createPinnedKey` or `getPinnedKey`. You could even go as far as calling it `usePinnedKey`, name it as a hook, which will make the code very easy to read later where you use this.

Pavel910 reviewed (changes requested):
Just a tiny change on exported function name, and we're good to go 👍

brunozoric commented on packages/app-admin-ui/src/Navigation/PinnedMenuItems.tsx:
```diff
@@ -0,0 +1,110 @@
+ * <PinnedMenuItems menuItems={menus} />
+ */
+export const PinnedMenuItems = ({ menuItems }: PinnedMenuItemsProps) => {
+    const pinnableMenus = useMemo(() => getPinnableMenus(menuItems), [menuItems]);
```
I would say that most of these useMemo are not necessary. You can just get everything you need inside last `useMemo` (`pinnedItems`).
What you cant get inside of that useMemo (local storage), fetch outside.

ku5ic committed:
feat(admin-ui): show parent icon for pinned menu items

Pinned menu items now display the icon of their parent menu, improving
visual consistency and navigation clarity. Added logic to extract and
render parent icons in the pinned section.

ku5ic committed:
refactor(navigation): move icon prop to menu element

Removes the `icon` prop from `PinnableMenuItem` and instead passes the icon
directly to the menu element via `React.cloneElement`. This simplifies the
component API and ensures icons are handled consistently within menu items.

ku5ic committed:
fix(admin-ui): handle possible undefined menu element

Ensure that the menu item's element is always defined before cloning it.
This prevents runtime errors when rendering pinned menu items with missing
elements. The non-null assertion improves type safety and UI stability.

ku5ic committed:
feat(ui): hide pinned menu items when sidebar is collapsed

Pinned menu items and their groups are now hidden when the sidebar is not
pinned. The pin icon's visibility also depends on both the menu item's and
sidebar's pinned state, improving UI consistency.

ku5ic committed:
refactor(navigation): rename PINNED_KEY to pinnedKey

Renamed the PINNED_KEY function to pinnedKey in PinnableMenuItem and updated
all references in related files for consistency. This improves naming
conventions and code readability in the navigation components.

ku5ic committed:
fix(admin-ui): improve pinned menu sorting logic

Pinned menu items now sort with missing items placed at the end, preventing
incorrect ordering when items are not found in the pinned order array.
Removed unnecessary useMemo for pinnableMenus and pinnableKeys to simplify
component logic.

brunozoric commented on packages/app-admin-ui/src/Navigation/PinnableMenuItem.tsx:
```diff
@@ -21,7 +21,7 @@ type PinnableMenuItemProps = {
  * @returns The localStorage key string for the pinned state.
  */
-export const PINNED_KEY = (name: string) => `navigation/${name}/pinned`;
+export const pinnedKey = (name: string) => `navigation/${name}/pinned`;
```
`const key = createPinnedKey(whatever)`

ku5ic committed:
refactor(nav): optimize pinned menu items memoization

Refactored PinnedMenuItems to memoize pinnableMenus, pinnableKeys, and
pinnedOrder together, reducing redundant computations and improving
performance. This change ensures dependencies are managed efficiently
and state updates are more predictable.

ku5ic committed:
feat(ui): rename sidebar state from pinned to expanded

Refactored sidebar state variable from `pinned` to `expanded` in navigation
components for improved clarity. Updated related conditional rendering and
class logic to use the new naming convention.

ku5ic committed:
refactor(navigation): rename pinnedKey to createPinnedKey

Renamed the `pinnedKey` function to `createPinnedKey` in navigation components
for improved clarity and consistency. Updated all references in related files.
No functional changes were made.

ku5ic committed:
Merge branch 'next' into ku5ic/feature/WEB-5364/menu_items_pinning_functionality

ku5ic committed:
refactor(nav): simplify pinned menu rendering logic

Removed sidebar expansion checks and conditional `Menu.Group` rendering
from `PinnedMenuItems`. Introduced `Separator` for clearer UI structure.

ku5ic committed:
refactor(properties): remove unused pinnable prop

The `pinnable` prop was removed from the `PropertyProps` interface and
related usages in the `Property` component. This prop was not being used
and its removal simplifies the component's API and internal logic.

ku5ic committed:
fix(admin-ui): simplify icon rendering for pinned menu items

Refactored PinnedMenuItems to use getParentIcon directly when rendering
icons for pinned menu items, removing unnecessary renderIcon function.
This streamlines the code and avoids redundant React.cloneElement calls.

ku5ic committed:
feat(ui): show pin icon on menu item hover

Enhances the PinnableMenuItem component to display the pin icon when the
parent menu item is hovered. Uses group-hover to improve user experience
by making the pin action more discoverable and accessible.

ku5ic committed:
feat(navigation): add parent icon context to menu items

Introduced MenuParentContext and useMenuParentIcon hook to provide parent menu
icons to navigation components. DecoratedMenu now renders menu items and links
with parent icons when available. Updated PinnedMenuItems to wrap elements in
MenuParentContext.Provider for icon propagation.

ku5ic committed:
feat(admin-ui): refactor menu decorators for parent icon

Refactored the menu item and link decorators to use explicit prop types
and improved readability by removing generic decorator logic. Now uses
SidebarMenuItemButtonProps and SidebarMenuItemLinkProps for type safety.
Display names for decorated components are set for easier debugging.

ku5ic committed:
feat(admin-ui): simplify menu decorators for parent icon

Refactored Menu.Item and Menu.Link decorators to use direct function
declarations instead of intermediate variables. Removed unnecessary
displayName assignments for cleaner code. This improves readability and
maintainability of the DecoratedMenu component.

ku5ic committed:
feat(cms): add wby-size-md class to menu icon

Adds the "wby-size-md" CSS class to the FontAwesomeIcon in the menu group
icon component for consistent sizing and styling.

Pavel910 committed:
Merge branch 'next' into ku5ic/feature/WEB-5364/menu_items_pinning_functionality

ku5ic committed:
feat(cms): prevent icon shrinking in menu items

Add `wby-shrink-0` class to menu icons to ensure they do not shrink when
displayed. This improves visual consistency in the admin menu layout.

ku5ic committed:
fix(ui): add padding around pinned menu separator

Wrapped the Separator component in a div with padding classes to improve
visual spacing between pinned menu items in the admin UI navigation.

ku5ic committed:
Merge branch 'next' into ku5ic/feature/WEB-5364/menu_items_pinning_functionality

ku5ic committed:
fix(sidebar): improve subitem action accessibility

Replaced IconButton with Icon for chevron in SidebarMenuSubItem to simplify
markup and improve accessibility. Disabled pointer events on the action
container in SidebarMenuSubButton to prevent unintended interactions.

adrians5j reviewed (commented):
Did some manual testing, and notice a couple of things.

#### 1. Labels missing

<img width="1508" height="1066" alt="Image" src="https://github.com/user-attachments/assets/0f8c8769-b01a-4fef-af0f-d7da6d8a5080" />

#### 2. Icon size incorrect

<img width="1668" height="847" alt="Image" src="https://github.com/user-attachments/assets/53f1060a-58c2-4c87-ad41-e68b7906ae99" />

#### 3. Extra Line 

<img width="1656" height="914" alt="Image" src="https://github.com/user-attachments/assets/1fd7510a-be84-49c3-ae8e-e0e41dcac3f8" />

#### 4. Smaller Icon Here

<img width="595" height="762" alt="Image" src="https://github.com/user-attachments/assets/840ebb3c-58c2-434f-929d-d957b8c9f8ac" />

#### 5. Pinned Section - Unpin Icon Should Behave Differently

<img width="1773" height="1127" alt="Image" src="https://github.com/user-attachments/assets/dd4a76fc-9a30-4d54-87ce-b5650145c626" />


--

Apart from visual stuff, I also think we should've tried to make the pin/unpin code more part of the `admin-ui/src/sidebar` component itself. 

Making the pinning feature more localised within the Sidebar component would make things simpler I think. One example I can give is fixing this bug that I just discovered. 

I just tried this:

```tsx
        <Menu.Link
                                action={<Menu.Link.Action element={<MoreIcon />} />}
                                text={"Redirects"}
                                to={router.getLink(Routes.Redirects.List)}
                            />
```

As we can see, I added an action to the link. Since the item also has `pinnable=true`, we have this visual bug, where two icons overlap:

<img width="501" height="614" alt="Image" src="https://github.com/user-attachments/assets/6c4621f8-c54c-452e-9f10-c8eab443aad4" />

Of course, providing the `onPin` / `onUnpin` callbacks and `pinnedItems` array should happen still from `app-admin` package.

Maybe we should discuss this before tackling the above?

ku5ic committed:
feat(ui): increase chevron icon size in sidebar subitem

Changed the chevron icon size from "xs" to "sm" in SidebarMenuSubItem for
better visibility and improved user experience in the sidebar navigation.

ku5ic committed:
feat(ui): add size prop to Pin icon in menu item

Adds the `size="sm"` prop to the Pin icon in `PinnableMenuItem` for improved
visual consistency and alignment with other UI elements.

ku5ic committed:
feat(admin-ui): add unpin icon to pinnable menu item

Replaced static pin icon with dynamic unpin icon when menu item is pinned.
Removed sidebar expansion dependency for icon visibility. Improves user
feedback and interaction clarity in the admin UI navigation.

ku5ic commented:
> Did some manual testing, and notice a couple of things.
> #### 1. Labels missing
> <img alt="Image" width="1508" height="1066" src="https://private-user-images.githubusercontent.com/5121148/500867445-0f8c8769-b01a-4fef-af0f-d7da6d8a5080.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA0MzYwNjUsIm5iZiI6MTc2MDQzNTc2NSwicGF0aCI6Ii81MTIxMTQ4LzUwMDg2NzQ0NS0wZjhjODc2OS1iMDFhLTRmZWYtYWYwZi1kN2RhNmQ4YTUwODAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMTRUMDk1NjA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGFlNDAxYjMxYTlmMjQ5ZjQzN2EzNTg4NGVlYTdmNWMxMDMwN2EwYTRiZDA1YTg4YzEzZDdhOTM2OWNlODNlOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.VNwhHYu2cX28JGnNKKt8Jjy27edOQdJZGjxDM0b62k0">
> #### 2. Icon size incorrect
> <img alt="Image" width="1668" height="847" src="https://private-user-images.githubusercontent.com/5121148/500867878-53f1060a-58c2-4c87-ad41-e68b7906ae99.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA0MzYwNjUsIm5iZiI6MTc2MDQzNTc2NSwicGF0aCI6Ii81MTIxMTQ4LzUwMDg2Nzg3OC01M2YxMDYwYS01OGMyLTRjODctYWQ0MS1lNjhiNzkwNmFlOTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMTRUMDk1NjA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmZjNTM2NDg0NDY2MjQwY2NjMDIyMGU1YzYwNTVmYmVkNmRiNmRjZjMyZDBiZmI3MDRmYzczOTk5NDEwYWUzMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.tcT2CKSUxH06fRAwtwjs5q2i5UDkLTHxI1dQd6mjxtM">
> #### 3. Extra Line
> <img alt="Image" width="1656" height="914" src="https://private-user-images.githubusercontent.com/5121148/500868687-1fd7510a-be84-49c3-ae8e-e0e41dcac3f8.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA0MzYwNjUsIm5iZiI6MTc2MDQzNTc2NSwicGF0aCI6Ii81MTIxMTQ4LzUwMDg2ODY4Ny0xZmQ3NTEwYS1iZTg0LTQ5YzMtYWU4ZS1lMGU0MWRjYWMzZjgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMTRUMDk1NjA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmMyZjUwMmU0ODAwMjM3YWJkNzdhZGFiMzIwMDY5YzJmMjBhMTZkZDJkNmZmN2IwZTQ4MDU4NGVjOWRhNjEyYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.9cEoUT4R9nVrt4bZ7Ys8jtxuZQaSKeI6FWvRdn-mnoI">
> #### 4. Smaller Icon Here
> <img alt="Image" width="595" height="762" src="https://private-user-images.githubusercontent.com/5121148/500869255-840ebb3c-58c2-434f-929d-d957b8c9f8ac.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA0MzYwNjUsIm5iZiI6MTc2MDQzNTc2NSwicGF0aCI6Ii81MTIxMTQ4LzUwMDg2OTI1NS04NDBlYmIzYy01OGMyLTQzNGYtOTI5ZC1kOTU3YjhjOWY4YWMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMTRUMDk1NjA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjYxOGNmN2NjYTUwOGQ1ZDgxZDUyMGNhZmVhZmJlZWFmMzY3MTM3YjM5OGM3ZWFkOWJhNTI0MzJkM2ZjNDk1YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.22PTEOhfg2X6oUiVrKEzcBeDvhDbi5a2g7gy3IP9wy0">
> #### 5. Pinned Section - Unpin Icon Should Behave Differently
> <img alt="Image" width="1773" height="1127" src="https://private-user-images.githubusercontent.com/5121148/500870604-dd4a76fc-9a30-4d54-87ce-b5650145c626.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA0MzYwNjUsIm5iZiI6MTc2MDQzNTc2NSwicGF0aCI6Ii81MTIxMTQ4LzUwMDg3MDYwNC1kZDRhNzZmYy05YTMwLTRkNTQtODdjZS1iNTY1MDE0NWM2MjYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMTRUMDk1NjA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzFjMjhkODJlYTMxMWVjYTA3ODRjZDk2M2E4ZTc1NmI4NzYzMDE2MzlmMWVjZTgyNTk2NmFmNmIxYmNjMzUxZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.uI-EJVw2uSSKt-tOd3chB9OYsRcfiGUajsLm32GRiSc">
> 
> --
> 
> Apart from visual stuff, I also think we should've tried to make the pin/unpin code more part of the `admin-ui/src/sidebar` component itself.
> 
> Making the pinning feature more localised within the Sidebar component would make things simpler I think. One example I can give is fixing this bug that I just discovered.
> 
> I just tried this:
> 
> ```tsx
>         <Menu.Link
>                                 action={<Menu.Link.Action element={<MoreIcon />} />}
>                                 text={"Redirects"}
>                                 to={router.getLink(Routes.Redirects.List)}
>                             />
> ```
> 
> As we can see, I added an action to the link. Since the item also has `pinnable=true`, we have this visual bug, where two icons overlap:
> <img alt="Image" width="501" height="614" src="https://private-user-images.githubusercontent.com/5121148/500874076-6c4621f8-c54c-452e-9f10-c8eab443aad4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjA0MzYwNjUsIm5iZiI6MTc2MDQzNTc2NSwicGF0aCI6Ii81MTIxMTQ4LzUwMDg3NDA3Ni02YzQ2MjFmOC1jNTRjLTQ1MmUtOWYxMC1jOGVhYjQ0M2FhZDQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMTRUMDk1NjA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTUwZGI1MjliNDRkMzM5ODc0MDg0ODYyNjA1MzIwM2ZmODE2ZGI1NTI0NmZiOGYxZGRkMGY2ODcwNTA4ODkxYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ZmzyciuZGLbTPyPWQ9a-Rzh_le909tiEzZ-BnD_09Wk">
> 
> Of course, providing the `onPin` / `onUnpin` callbacks and `pinnedItems` array should happen still from `app-admin` package.
> 
> Maybe we should discuss this before tackling the above?

Just to summarize here for reference:

1) Labels missing: required by @SvenAlHamad offline
1) Icon size incorrect: corrected :white_check_mark:
1) Extra Line: required by @SvenAlHamad offline
1) Smaller Icon Here: corrected :white_check_mark:
1) Pinned Section - Unpin Icon Should Behave Differently: corrected :white_check_mark:
1) Bug/Refactor: Let's wait for a team input, especially @Pavel910   :grey_question:

ku5ic committed:
Merge branch 'next' into ku5ic/feature/WEB-5364/menu_items_pinning_functionality

adrians5j reviewed (approved):
From my side, LGTM 🤝 🙂
